api/src/test/java/org/openmrs/module/datafilter/impl/UserProgramBasedFilterTest.java

```diff
@@ -116,4 +128,18 @@ public void getUsers_shouldReturnAllUsersIfTheFilterIsDisabled() {
 		assertEquals(7, getUsers().size());
 	}
 	
+	@Test(expected = None.class)
+	public void getProvidersByPerson_shouldNotFailWithSQLSyntaxErrorException() throws Exception {
+		// Setup
+		String sql = "DELETE FROM datafilter_entity_basis_map";
+		FilterTestUtils.enableFilter(ImplConstants.PROGRAM_BASED_FILTER_NAME_USER);
+		DatabaseUtil.executeSQL(getConnection(), sql, false);
```

Comment:
Why are you calling this?

Reply:
I have to make it into [this block above](https://github.com/openmrs/openmrs-module-datafilter/pull/19/files/032c855356ae5450339a577eaf81eab41588df4f#diff-9131b4706b019c82d4199ff859a46cca876cdf721da3f24b94c8a24cc6cae614R105). `if (allProgramRoleNames.isEmpty()) {...)`.

Reply:
ok